src/Transport/Topology/Topologies/ForwardingTopology.cs

```diff
@@ -109,6 +110,14 @@ public Func<IDispatchMessages> GetDispatcherFactory()
             return () => container.Resolve<IDispatchMessages>();
         }
 
+        public StartupCheckResult ApplyPreStartupChecks()
+        {
+            var checks = container.ResolveAll<IStartupCheck>();
+            var compositeStartupChecker = new CompositeStartupCheck(checks);
```

Comment:
> In my opinion we could give to users the possibility to add "pre startup checks" 

You do that by allowing custom topology to implement a pre-startup check.
Beyond that it feels as if we're exposing infrastructure hooks. Does it belong here?